check_clang_tidy.sh

```diff
@@ -16,7 +16,27 @@ _travis_run_clang_tidy_fix() {
     for pkg in ${SOURCE_PKGS[@]} ; do  # process files in topological order
         test -z "${PKGS[$pkg]}" && continue  # skip pkgs without compile_commands.json
         travis_fold start clang.tidy "  - $(colorize BLUE Processing $pkg)"
-        travis_run_wait "$RUN_CLANG_TIDY_EXECUTABLE -fix -p ${PKGS[$pkg]} 2> /dev/null"
+
+        # Find all .cpp files in pkg's src_dir that were added or modified in this pull request
+        # If we are not processing a Travis pull request, check all files
+        # To enable unit-testing, check all files when $TEST_PKG is defined
+        modified_files=()
+        if [ "${TRAVIS_PULL_REQUEST:-false}" != false ] && [ -z "$TEST_PKG" ] ; then
+            src_dir=$(grep "^CMAKE_HOME_DIRECTORY:INTERNAL=" "${PKGS[$pkg]}/CMakeCache.txt")
+            src_dir=$(realpath "${src_dir#*=}")
+            strip_prefix=$(cd "$src_dir"; git rev-parse --show-toplevel)
```

Comment:
could you add a comment about what this line is doing? rev-parse.... "git-rev-parse - Pick out and massage parameters"?

Reply:
Done